src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestGHEventSubscriber.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.cloudbees.jenkins.GitHubRepositoryName;
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.security.ACL;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.SCMSourceOwner;
+import jenkins.scm.api.SCMSourceOwners;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.github.extension.GHEventsSubscriber;
+import org.kohsuke.github.GHEvent;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.Sets.immutableEnumSet;
+import static org.kohsuke.github.GHEvent.PULL_REQUEST;
+
+/**
+ * This subscriber manages {@link org.kohsuke.github.GHEvent} PULL_REQUEST.
+ */
+@Extension
+public class PullRequestGHEventSubscriber extends GHEventsSubscriber {
+
+    private static final Logger LOGGER = Logger.getLogger(PullRequestGHEventSubscriber.class.getName());
+    private static final Pattern REPOSITORY_NAME_PATTERN = Pattern.compile("https?://([^/]+)/([^/]+)/([^/]+)");
+
+
+    @Override
+    protected boolean isApplicable(@Nullable Job<?, ?> project) {
+        return true;
```

Comment:
should check the job is interested in such events. Or it will register hooks for every job and just skipped after such event occurs